### ISSUE TITLE
logging the learning rates during training 

### ISSUE THREAD

**Issue by shane-huang** [intel-analytics/BigDL#257]
Printing learning rate (and/or other hyper params) in logs is of great help for DL application development/debugging. This is the default behavior of both Torch and Caffe. 

In our current implementation, learning rate (along with other hyper params) is updated inside optimMethod (i.e. SGD). DistOptimizer is unaware of any changes of hyper parameters and thus unable to output any meaningful logs for debugging.  

We could either pass the logger into optimMethod for certain status output, or make optimize() function return a structure indicating the status.  If there're concerns about too many logs, we could put those logs in optimMethod in debug mode.

**Comment by qiuxin2012** [comment]
@shane-huang I put learning rate to the debug logs. You can get the output like this, is this OK?

```
2017-01-18 13:15:18 INFO  DistriOptimizer$:242 - [Epoch 1 192/1024][Iteration 7][Wall Clock 0.181247672s] Train 32 in 0.02696378seconds. Throughput is 1186.777224854972 records/second. Loss is 0.2614013580068211. 
2017-01-18 13:15:18 DEBUG DistriOptimizer$:246 - 
========== Metrics Summary ==========
get weights average : 1.7414E-4 s
computing time average : 1.70687E-4 s
aggregate gradient time : 5.581975E-5 s
get weights for each node : 1.97537E-4 1.17383E-4 1.23858E-4 2.57782E-4 
computing time for each node : 1.89211E-4 1.0787E-4 2.08731E-4 1.76936E-4 
=====================================
2017-01-18 13:15:18 DEBUG DistriOptimizer$:247 - Dropped modules: 0
2017-01-18 13:15:18 DEBUG DistriOptimizer$:256 - Current learning rate is 0.001
```
